### PR TITLE
Various improvements to wstETH adapter

### PR DESCRIPTION
**Pull request by MerlinEgalite**
Fixes:
- https://github.com/cantinasec/review-morpho/issues/8
- https://github.com/cantinasec/review-morpho/issues/7
- https://github.com/cantinasec/review-morpho/issues/11

### CODE REVIEW

**Review by Rubilmax — APPROVED** [review]

File: src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol [hunk 1]
```diff
@@ -4,16 +4,22 @@ pragma solidity 0.8.21;
...
+/// @dev This contract should only be used as price feed for `MorphoChainlinkOracleV2`.
+contract WstEthStEthExchangeRateChainlinkAdapter is MinimalAggregatorV3Interface {
+    /// @inheritdoc MinimalAggregatorV3Interface
     uint8 public constant decimals = 18;
-    string public constant description = "wstETH/ETH exchange rate";
+
+    /// @notice The description of the price feed.
+    string public constant description = "wstETH/stETH exchange rate";
+
+    /// @notice The address of stETH on Mainnet.
```
Comment (MathisGD) [inline src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol:19]:
```suggestion
    /// @notice The address of stETH on Ethereum.
```

File: src/wsteth-exchange-rate-adapter/interfaces/MinimalAggregatorV3Interface.sol [hunk 2]
```diff
@@ -3,9 +3,13 @@ pragma solidity >=0.5.0;
 
 /// @dev Inspired by
 /// https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol
+/// @dev This is the minimal feed interface required by `MorphoChainlinkOracleV2`.
 interface MinimalAggregatorV3Interface {
+    /// @notice Returns the number of decimals precision.
```
Comment (MathisGD) [inline src/wsteth-exchange-rate-adapter/interfaces/MinimalAggregatorV3Interface.sol:8]:
```suggestion
    /// @notice Returns the number of decimals of precision of the feed.
```
or
```suggestion
    /// @notice Returns the precision of the feed.
```

File: src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol [hunk 3]
```diff
@@ -4,16 +4,22 @@ pragma solidity 0.8.21;
...
     IStEth public constant ST_ETH = IStEth(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
 
+    /// @inheritdoc MinimalAggregatorV3Interface
```
Comment (MathisGD) [inline src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol:22]: I think that "Only the `answer` field is used by `MorphoChainlinkOracleV2`." isn't clear enough on the fact that the other values returned here are non-sense. I would add:
```suggestion
    /// @inheritdoc MinimalAggregatorV3Interface
    /// @dev Returns zero for roundId, startedAt, updatedAt and answeredInRound.
```

File: src/wsteth-exchange-rate-adapter/interfaces/MinimalAggregatorV3Interface.sol [hunk 4]
```diff
@@ -3,9 +3,13 @@ pragma solidity >=0.5.0;
...
     function decimals() external view returns (uint8);
 
+    /// @notice Returns Chainlink's `latestRoundData` return values.
```
Comment (MathisGD) [inline src/wsteth-exchange-rate-adapter/interfaces/MinimalAggregatorV3Interface.sol:11]: not super useful

**Review by MathisGD — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

File: src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol [hunk 5]
```diff
@@ -4,16 +4,23 @@ pragma solidity 0.8.21;
...
 
-/// @title WstEthEthExchangeRateChainlinkAdapter
+/// @title WstEthStEthExchangeRateChainlinkAdapter
 /// @author Morpho Labs
 /// @custom:contact security@morpho.org
-/// @notice wstETH/ETH exchange rate price feed.
-/// @dev This contract should only be used as price feed for `ChainlinkOracle`.
-contract WstEthEthExchangeRateChainlinkAdapter is MinimalAggregatorV3Interface {
+/// @notice wstETH/stETH exchange rate price feed.
+/// @dev This contract should only be deployed on Ethereum and used as price feed for `MorphoChainlinkOracleV2`.
```
[the remark this answers is not in the document]
Comment (Rubilmax) [reply src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol:11]:
```suggestion
/// @dev This contract should only be deployed on Ethereum and used as a price feed for `MorphoChainlinkOracleV2`.
```

**Review by Rubilmax — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Comment by StErMi** [comment]
I'm wondering if it makes sense to drop everywhere the reference to Chainlink everywhere and just use the "Morpho" terms. Like Morpho Adapter, Morpho Oracle and so on. Those will be "inspired" to the Chainlink version in the sense that you (Morpho) have agreed that their interface is a "standard de facto" for the function signatures for an oracle.

**Review by MerlinEgalite — COMMENTED** [review]
LGTM

**Review by Rubilmax — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

**Pull request merged**